Lib/test/test_compileall.py

```diff
@@ -55,9 +55,25 @@ def timestamp_metadata(self):
         with open(self.bc_path, 'rb') as file:
             data = file.read(12)
         mtime = int(os.stat(self.source_path).st_mtime)
-        compare = struct.pack('<4sll', importlib.util.MAGIC_NUMBER, 0, mtime)
+        compare = struct.pack('<4sLL', importlib.util.MAGIC_NUMBER, 0,
+                              mtime & 0xFFFF_FFFF)
         return data, compare
 
+    def test_year_2038_mtime_compilation(self):
+        # Test to make sure we can handle mtimes larger than what a 32-bit
+        # signed number can hold as part of bpo-34990
+        os.utime(self.source_path, (2**32 - 1, 2**32 - 1))
```

Comment:
I expect that some filesystems may fail on date arounds the 32-bit epoch overflow. Maybe OSError should be catched here as well, as done in test_larger_than_32_bit_times?

Reply:
Better safe than sorry, some OSs might be using signed 32-bit numbers for timestamps.